.github/workflows/test-python-ci.yml

```diff
@@ -0,0 +1,120 @@
+      - .github/workflows/test-python-ci.yml
+  push:
+    branches:
+      - master
+
+env:
+  rust-version: 1.62.0
+  poetry-version: 1.2.0b3
+  postgre-version: 12
+  python-version: '3.9'
+  pytest-base-args: >-
+    --log-level=DEBUG
+    --durations=10
+    --side-effects-timeout=10
+    -vv
+    -x
+jobs:
+  test-python-matrix:
+    strategy:
+      fail-fast: false
+      matrix:
+        os:
+          - ubuntu-20.04
+          - macos-12
+          - windows-2022
+    runs-on: ${{ matrix.os }}
+    steps:
+        # Install required packages for ubuntu:
+        # - LibFuse2 since only libfuse3 is available on ubuntu-20.04,
+        # - `desktop-file-utils` provides `update-desktop-database` used by `tests/scripts/run_testenv.sh`
+        # - A specific version of postgresql is used for postgre's tests
+        # - dependencies for Qt testing. See: https://pytest-qt.readthedocs.io/en/latest/troubleshooting.html#github-actions
+      - name: (Linux) Install packages fuse2, postgre-${{ env.postgre-version }}
+        if: startsWith(matrix.os, 'ubuntu-')
+        run: >
+          sudo apt-get -y install
+          libfuse2
+          fuse
+          desktop-file-utils
+          postgresql-${{ env.postgre-version }}
+          libxkbcommon-x11-0 libxcb-icccm4 libxcb-image0
+          libxcb-keysyms1 libxcb-randr0 libxcb-render-util0
+          libxcb-xinerama0 libxcb-xfixes0 x11-utils
+
+      - name: (MacOS) Install macfuse
+        if: startsWith(matrix.os, 'macos-')
+        run: brew install --cask macfuse
+
+      - name: (Windows) Install WinFSP
+        if: startsWith(matrix.os, 'windows-')
+        shell: bash
+        run: |
+          choco install -y --limit-output winfsp --version=${{ env.winfsp-version }}
+          mkdir winfsp-test
+          curl -L https://github.com/billziss-gh/winfsp/releases/download/v1.8/winfsp-tests-1.8.20304.zip -o winfsp-test/winfsp-tests.zip
+          unzip winfsp-test/winfsp-tests.zip -d winfsp-test
+          pwd
+          echo "$(pwd)\winfsp-test" >> "$GITHUB_PATH"
+
+      - uses: actions/checkout@2541b1294d2704b0964813337f33b291d3f8596b # pin v3.0.2
+
+      - uses: ./.github/actions/setup-rust
+        with:
+          version: ${{ env.rust-version }}
+          cache-key: python
+        timeout-minutes: 10
+
+      - uses: ./.github/actions/setup-python-poetry
+        id: setup-python
+        with:
+          python-version: ${{ env.python-version }}
+          poetry-version: ${{ env.poetry-version }}
+
+      - name: Install python deps
+        run: poetry install -E core -E backend
+
+      - name: Debug python & pip version
+        shell: bash
+        continue-on-error: true
+        run: |
+          set -x
+          poetry run python --version
+          poetry run pip --version
+
+      - name: Basic tests
+        run: poetry run pytest ${{ env.pytest-base-args }} tests -n auto
+        timeout-minutes: 5
+
+      - name: (Not MacOS) Mountpoint tests
```

Comment:
*Sad macos noise*

More seriously, we currently don't run the mountpoint tests on macos ? (This seems rather troublesome!)

Reply:
No, and If we want to that's for another PR